File: sdk/rust/examples/push.rs
use oomclient::{Client, Value};

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let mut client = Client::connect("http://localhost:50051").await?;

    let kvs = [
        ("last_5_click_posts".to_string(), Value::String("hello".into())),
        ("number_of_user_starred_posts".to_string(), Value::Int64(28)),
    ]
    .into_iter()
    .collect();

    client.push("1", "user-click", kvs).await?;

    Ok(())
}

File: sdk/python/examples/push.py
#!/usr/bin/env python3
import asyncio
from oomclient import Client


async def main():
    client = await Client.connect("http://localhost:50051")
    kv_pairs = {
        "last_5_click_posts": "1,3,5,7,9",
        "number_of_user_starred_posts": 30,
    }
    await client.push("user-click", "929", kv_pairs)


asyncio.run(main())
